File: Library.Core.Web/src/app/view-models/Book/postBookView.ts
export class PostBookView {
  constructor(
    public id: number = 0,
    public name?: string,
    public yearOfPublishing: number = 0,
    public authors?: AuthorPostBookViewItem[],
    public publicationHouses?: PublicationHousePostBookViewItem[]
  ) { }
}

export class AuthorPostBookViewItem {
  constructor(
    public id: number = 0,
    public name?: string,
    public birthday: number = 0,
    public deathday: number = 0
  ) { }
}

export class PublicationHousePostBookViewItem {
  constructor(
    public id: number = 0,
    public name?: string,
    public adress?: string
  ) { }
}

File: Library.Core.ViewModelLayer/ViewModels/BookViewModels/PostBookView.cs
using System.Collections.Generic;

namespace Library.Core.ViewModelLayer.ViewModels.Book
{
    public class PostBookView
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int YearOfPublishing { get; set; }

        public ICollection<AuthorPostBookViewItem> Authors { get; set; }
        public ICollection<PublicationHousePostBookViewItem> PublicationHouses { get; set; }

        public PostBookView()
        {
            Authors = new List<AuthorPostBookViewItem>();
            PublicationHouses = new List<PublicationHousePostBookViewItem>();
        }
    }

    public class AuthorPostBookViewItem
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int Birthday { get; set; }
        public int Deathday { get; set; }

        public AuthorPostBookViewItem()
        {

        }
    }

    public class PublicationHousePostBookViewItem
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Adress { get; set; }

        public PublicationHousePostBookViewItem()
        {

        }
    }
}
